core-lib/TestSuite/LiteralObjectTests.som

```diff
@@ -0,0 +1,102 @@
+of this software and associated documentation files (the 'Software'), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*)
+class ObjectTests usingPlatform: platform testFramework: minitest = (
+| private TestContext = minitest TestContext. |
+) (
+  public class ObjectTest = TestContext ()(
+
+    public testEquality = (
+      | a b |
+      a := objL()().
+      b := objL()().
+      self deny: a == b.
+    )
+  
+    public testMethodLocalRead = (
+      | x a |
+      x := 1.
+      a := objL()(public b = (^ x)).
+      self assert: a b = 1.
+    )
+
+    public testMethodLocalWrite = (
+      | x a |
+      x := 1.
+      a := objL()(public b = (x := x + 1. ^ x)).
+      self assert: a b = 2.
+    )
+    
+    public testBlockLocalRead = (
+      | x |
+      x := [ | z | z := 1. objL()(public b=(^ z)) b. ] value.
+      self.assert: x = 1.
+    )
+
+    public testBlockLocalWrite = (
+      | x |
+      x := [ :y | | z | objL()(public b=(z := 2. ^y + z)) b. ] value: 1.
+      self assert: x = 3. 
+    )
+
+    public testMethodLocalReadInBlock = (
+      | x a |
+      x := 1.
+      a := objL()(public b = (^ x)).
+      self assert: [ a b ] value = 1.
+    )
+
+    public testMethodLocalWriteInBlock = (
+      | x a |
+      x := 1.
+      a := objL()(public b = (x := x + 1. ^ x)).
+      self assert: [ a b ] value = 2.
+    )
+    
+    public testMethodLocalReadWhenNested = (
+      | x y |
+      x := 1.
+      y := [ objL()(public b = (^ x)) b ] value.
+      self assert: y = 1.
+    )
+
+    public testMethodLocalWriteWhenNested = (
+      | x y |
+      x := 1.
+      y := [ objL()(public b = (x := x + 1. ^ x)) b ] value.
+      self assert: y = 2.
+    )
+
+    public testOuterKeywordInObjectLiteral = (
+      | x y |
+      x := objL()(public b = (^ outer ObjectTests)) b.
+      y := outer ObjectTests.
+      self assert: x == y.
+    )
+    
+    public testOuterKeywordInObjectLiteralInBlock = (
+      | x y |
+      x := [objL()(public b = (^ outer ObjectTests)) b] value.
+      y := outer ObjectTests.
+      self assert: x == y.
+    )
+    
```

Comment:
We'd also want to test giving a super class to the literal, and that the primary factory method initialize it correctly.